api/python/provisioner/commands/check.py

```diff
@@ -623,7 +723,14 @@ def run(self, check_name: cfg.Checks = None,
         res: CheckResult = CheckResult()
 
         if check_name is None or check_name == cfg.Checks.ALL.value:
-            for check_name in cfg.CHECKS:
+            check_lists = cfg.SWUPDATE_CHECKS
+        else:
+            try:
+                check_lists = getattr(cfg, check_name.strip().upper())
```

Comment:
In my opinion, it is better to refer to list checks by its names rather than try to obtain them from `config` module at the top level

Reply:
Simplified design by making 2 different classes for checks:
Getting directly list of checks from config using getattr

    """Enumeration for available checks/validations"""

    NETWORK = "network"
    CONNECTIVITY = "connectivity"
    BMC_ACCESSIBILITY = "bmc_accessibility"
    COMMUNICABILITY = "communicability"
    CLUSTER_STATUS = "cluster_status"
    LOGS_ARE_GOOD = "logs_are_good"
    PASSWORDLESS_SSH_ACCESS = "passwordless_ssh_access"
    STORAGE_LVMS = "storage_lvms"
    STORAGE_LUNS = "storage_luns"
    MGMT_VIP = "mgmt_vip"
    HOSTNAMES = "hostnames"


class GroupChecks(Enum):

    """Enum for group checks"""
    ALL = "all"
    POST_CHECKS = "post_checks"
    REPLACENODE_CHECKS = "replacenode_checks"
    SWUPDATE_CHECKS = "swupdate_checks"